examples/jetstream/queue-push-consumer/go/main.go

```diff
@@ -0,0 +1,159 @@
+	// part of a group, we can also create a core NATS subscription
+	// to join the group. As a resminder, the `DeliverSubject` is
+	// randomly generated by default, but this can be set explicitly
+	// in the consumer config if desired.
+	sub3, _ := nc.QueueSubscribeSync(info.Config.DeliverSubject, info.Config.DeliverGroup)
+	fmt.Printf("deliver subject: %q\n", info.Config.DeliverSubject)
+
+	// Now we can publish some messages to the stream to observe how
+	// they are distributed to the subscribers.
+	js.Publish("events.1", nil)
+	js.Publish("events.2", nil)
+	js.Publish("events.3", nil)
+	js.Publish("events.4", nil)
+	js.Publish("events.5", nil)
+	js.Publish("events.6", nil)
+
+	// As noted in the [push consumer][1] example, subscriptions
+	// enqueue messages proactively. When there are a group of
+	// subscriptions, each will receive a different subset of the messages.
+	// When calling `NextMsg` this means, messages can be processed out
+	// of order. There is no correlation with message order and
+	// subscription creation order 😉. In fact, it is possible that
+	// not all subscriptions will necessarily get a message.
+	// [1]: /examples/jetstream/push-consumer/go/
+	msg, _ := sub1.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub1: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub1: receive timeout")
+	}
+
+	msg, _ = sub2.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub2: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub2: receive timeout")
+	}
+
+	msg, _ = sub3.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub3: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub3: receive timeout")
+	}
+
+	// Like all ephemeral consumers, once the subscription that created
+	// it unsubscribes (or calls `Drain`), the consumer will be deleted.
+	sub1.Unsubscribe()
+	sub2.Unsubscribe()
+	sub3.Unsubscribe()
+
+	// ### Durable
+	// To create a (safe) durable consumer, use the `AddConsumer`
+	// method. Although it may seem redundant, a durable name *and*
+	// the deliver group name must be defined. This is simply because
+	// the durable name is used for all consumer types, while the
+	// deliver group is exclusive to the queue push consumer. In this
+	// example, the same name is used as convention.
+	fmt.Println("\n# Durable")
+
+	js.AddConsumer(streamName, &nats.ConsumerConfig{
+		Durable:        "event-processor",
+		DeliverSubject: "my-subject",
+		DeliverGroup:   "event-processor",
+		AckPolicy:      nats.AckExplicitPolicy,
+	})
+	defer js.DeleteConsumer(streamName, "event-processor")
+
+	// Wait for all 6 messages to be received before exiting.
+	wg := &sync.WaitGroup{}
+	wg.Add(6)
+
+	// For this part, we will use async core NATS queu subscriptions.
+	sub1, _ = nc.QueueSubscribe("my-subject", "event-processor", func(msg *nats.Msg) {
+		fmt.Printf("sub1: received message %q\n", msg.Subject)
```

Comment:
maybe we could mention that do not need to ack in QueueSubscribe by default since there is auto-ack, unlike with QueueSubscribeSync, or that could use nats.ManualAck option to disable auto acking

Reply:
Good catch, I also forget that the default behavior for async is auto-ack independent of the consumer config.. explicit ack which I feel like can be easily confused with requiring the subscriber to call `Ack` explicitly.. but it is for the _consumer_ to expect an ack.

Reply:
Hrm.. so since I am using `nc` to setup the subscription, I (obviously) can't pass `nats.ManualAck` nor I do presume that async handle knows that it should ack (since its not a JS API). In that case, I guess the manual `msg.Ack()` would be required.

Reply:
ah right since these are core NATS subscriptions then there is no auto ack either